Front/assets/modulos/Cart.js

```diff
@@ -49,4 +49,128 @@ class Productos() {
     };
 }
 
-*/
\ No newline at end of file
+*/
+
+/*
+let car = new Carrito(1);
+let contador = 0;*/
+
+
+
+class Cart {
+    constructor(clave) {
+        this.clave = clave;
+        this.productos = this.obtener();
+        this.ML_ENDPOINT= 'product';
+       
+        
+    }
+    agregar(producto) {
+            this.productos.push(producto);
+            localStorage.setItem(this.clave,JSON.stringify( this.productos));
+    }
+  /*  quitar(id) {
+        const indice = this.productos.findIndex(p => p.id === id);
+        if (indice != -1) {
+            this.productos.splice(indice, 1);
+            this.guardar();
+        }
+    }*/
+    obtener() {
+        if ( this.clave in localStorage){
+            return JSON.parse(localStorage.getItem(this.clave))
+        }else
+        return [];
+        
+    }
+
+    /*existe(id) {
+        return this.productos.find(producto => producto.id === id);
+    }*/
+
+    obtenerConteo() {
+        return this.productos.length;
+    }
+    
+    limpiar() {
+        this.productos=[];
+         localStorage.clear();
+         let listProducts = document.getElementById("listProducts");
+         listProducts.innerHTML = '';
+         let contCar = document.getElementById("contCar");
+         contCar.textContent = 0;
+         let totalCar = document.getElementById("totalCar");
+         totalCar.textContent = 0;
+    }
+
+    async mostrarProductos(){
+        let total=0;
+        for (const i in  this.productos) {
+                try {
+                    const response = await fetch(`http://127.0.0.1:3000/${this.ML_ENDPOINT}?q=${this.productos[i]}`);
```

Comment:
Why you think this is is needed (the call to ML)?, can the product detail be retrieved from local storage? or is it a bad idea?

Reply:
I tried to send a json on the button of each product but it didn't work, just send the id